src/mscorlib/src/System/String.cs

```diff
@@ -49,7 +49,7 @@ public sealed class String : IComparable, ICloneable, IConvertible, IEnumerable
 
         // For empty strings, this will be '\0' since
         // strings are both null-terminated and length prefixed
-        [NonSerialized] private char m_firstChar;
+        [NonSerialized] internal char m_firstChar;
```

Comment:
@jkotas I have changed the access modifier back to private and moved the actual work in the function to String.